openmetadata-ui/src/main/resources/ui/src/components/Glossary/GlossaryTermTab/GlossaryTermTab.component.tsx

```diff
@@ -249,6 +291,203 @@ const GlossaryTermTab = ({
     return data;
   }, [glossaryTerms, permissions]);
 
+  const listOfVisibleColumns = useMemo(() => {
+    return ['name', 'description', 'owners', 'status', 'new-term'];
+  }, [permissions]);
+
+  const defaultCheckedList = useMemo(() => {
+    return columns
+      .filter((column) => listOfVisibleColumns.includes(column.key as string))
+      .map((column) => column.key)
+      .filter((key): key is string => key !== undefined);
```

Comment:
@shrushti2000 , let's use `.reduce` here to have only one looping.

Reply:
Done. Refactored the code to use .reduce() as per suggestion. Now, the columns are processed in a single loop instead of chaining multiple .filter() and .map() calls.